### PR TITLE
Show format string completion options when the formatting component colon is typed

### PR DESCRIPTION
**Pull request by jnm2**
Same principle as https://github.com/dotnet/roslyn/pull/45309. Requested by @CyrusNajmabadi.

### CODE REVIEW

File: src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_DateAndTime.vb [hunk 1]
```diff
@@ -48,6 +48,28 @@ class c
...
+        <WpfTheory, CombinatorialData, Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function CompletionListIsShownWithDefaultFormatStringAfterTypingFormattingComponentColon(showCompletionInArgumentLists As Boolean) As Task
+            Using state = TestStateFactory.CreateCSharpTestState(
+                <Document><![CDATA[
+using System;
+class c
+{
+    void goo(DateTime d)
+    {
+        _ = $"Text {d$$}";
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_DateAndTime.vb:60]: test with `d:hh` and typing `:` after that.

Comment (jnm2) [reply src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_DateAndTime.vb:60]: Pushed.

**Comment by CyrusNajmabadi** [comment]
Awesome, thanks much!

**Comment by jnm2** [comment]
Sure!

**Pull request merged**